proto/umee/leverage/v1/tx.proto

```diff
@@ -83,15 +83,14 @@ message MsgRepay {
   cosmos.base.v1beta1.Coin asset    = 2 [(gogoproto.nullable) = false];
 }
 
-// MsgLiquidate represents a liquidator's request to repay a specific borrower's
-// borrowed base asset type to the module in exchange for collateral reward.
+// MsgLiquidate is the request structure for the Liquidate RPC.
 message MsgLiquidate {
   // Liquidator is the account address performing a liquidation and the signer
   // of the message.
-  string                   liquidator = 1;
-  string                   borrower   = 2;
-  cosmos.base.v1beta1.Coin repayment  = 3 [(gogoproto.nullable) = false];
-  cosmos.base.v1beta1.Coin reward     = 4 [(gogoproto.nullable) = false];
+  string                   liquidator   = 1;
+  string                   borrower     = 2;
+  cosmos.base.v1beta1.Coin repayment    = 3 [(gogoproto.nullable) = false];
+  string                   reward_denom = 4;
```

Comment:
In that case the complexity is moved to liquidator... he will need to calculate which denom he can liquidate and how much. 

We are already adjusting the repayment. Instead we can go one by one of the collateral list and sum app to the value related to the repayment. We could use denom list to make it simpler.

Reply:
Yes, the liquidator can do that - I had a script doing as much during the testnet, and what it does is proceed in order of preference for repay and reward denoms by offering the maximum amount it is willing to repay.

For most borrowers, even with multiple borrowed and collateral types, the first `MsgLiquidate` brings them back to health, so the liquidator's initial preference for reward and repay denoms is enough.

For borrowers where multiple transactions are needed, the liquidator only needs to follow their ranked order of preferences, because the first transaction will have either exhausted a repay denom or a reward denom by succeeding.

Note: Exhausted a denom can mean:
- Repaid a borrow completely in one token denom
- Reduced liquidator's token balance to zero (or consumed maximum offered amount) for a token denom
- Reduced borrower's collateral position in a uToken denom to zero
- Exhausted available supply of a reward token

Reply:
Liquidation logic is independent of final state (except for the collateral == 0 check for bad debt flagging) so that isn't a problem.

In particular, the test case labeled `repayLimited` covers this scenario in `liquidate_test.go`